userspace/falco/app/actions/configure_interesting_sets.cpp

```diff
@@ -117,6 +123,25 @@ static void select_event_set(falco::app::state& s, const libsinsp::events::set<p
 		falco_logger::log(LOG_DEBUG, "-(" + std::to_string(user_negative_sc_set_names.size())
 			+ ") syscalls removed (base_syscalls override): "
 			+ concat_set_in_order(user_negative_sc_set_names) + "\n");
+		auto invalid_negative_sc_set_names = unordered_set_difference(user_negative_names, user_negative_sc_set_names);
+		if (!invalid_negative_sc_set_names.empty())
+		{
+			falco_logger::log(LOG_WARNING, "Invalid (negative) syscall names: warning (base_syscalls override): "
+				+ concat_set_in_order(invalid_negative_sc_set_names));
+		}
+	}
+
+	/* REPLACE DEFAULT STATE, nothing else. */
+	if (s.config->m_base_syscalls_repair && s.config->m_base_syscalls_custom_set.empty())
+	{
+		/* If `base_syscalls.repair` is specified, but `base_syscalls.custom_set` is empty we are replacing
+		 * the default `sinsp_state_sc_set()` enforcement with the alternative `sinsp_repair_state_sc_set`.
+		 * This approach only activates additional syscalls Falco needs beyond the
+		 * syscalls defined in each Falco rule that are absolutely necessary based
+		 * on the current rules configuration. */
+
+		// returned set already has rules_sc_set merged
+		s.selected_sc_set = libsinsp::events::sinsp_repair_state_sc_set(rules_sc_set);
```

Comment:
```suggestion
	/* REPLACE DEFAULT STATE, nothing else. */
	if (s.config->m_base_syscalls_repair && user_positive_sc_set.empty())
	{
		/* If `base_syscalls.repair` is specified, but `base_syscalls.custom_set` defines no positive syscall we are replacing
		 * the default `sinsp_state_sc_set()` enforcement with the alternative `sinsp_repair_state_sc_set`.
		 * This approach only activates additional syscalls Falco needs beyond the
		 * syscalls defined in each Falco rule that are absolutely necessary based
		 * on the current rules configuration. */

		base_sc_set.clear()
```
Agreed on your proposal. I think this should happen only when `user_positive_sc_set` is empty, so that users can still negate some syscall if they need to (potentially used by the rules for maximum control). Also, this needs to be moved at line 110 as an `else` branch of the current `if (!user_positive_sc_set.empty())`, so that negated syscall checks can be applied below.

What do you think?

Reply:
ACK, since `sinsp_repair_state_sc_set` already merges in the rules_sc_set moving it to line 115 may be better. I think it's ok to just re-assign / override s.selected_sc_set again, but we could add more if else, elseif etc behavior as well.